include/private/entry.h

```diff
@@ -148,14 +142,62 @@ struct stumpless_element *
 locked_get_element_by_index( const struct stumpless_entry *entry,
                              size_t index );
 
+/**
+ * Retrieves an element by name from a Stumpless entry, ensuring thread safety.
+ *
+ * This function searches for an element within the given entry by its name. 
+ * It locks each element during comparison to ensure thread safety. If the 
+ * element with the specified name is found, it is returned.
+ *
+ * **Thread Safety: MT-Safe**
+ * This function is thread-safe as it locks individual elements during the search process.
+ *
+ * **Async Signal Safety: AS-Unsafe**
+ * Unsafe to call from asynchronous signal handlers due to lock manipulation.
+ *
+ * **Async Cancel Safety: AC-Unsafe**
+ * Not safe in contexts of asynchronous cancellation, as it might involve lock manipulation.
+ *
+ * @since release 2.0.0
```

Comment:
Don't add `@since` tags in your additions - it's unnecessary work to go find the release that private functions were added in. If they're already present you can leave them, but don't worry about adding them to new documentation.